pkg/resources/external_oauth_integration.go

```diff
@@ -41,548 +46,777 @@ var oauthExternalIntegrationSchema = map[string]*schema.Schema{
-			ExternalOauthAllowedRolesList:                expandStringList(d.Get("allowed_roles").(*schema.Set).List()),
-			ExternalOauthAllowedRolesListOk:              isOk(d.GetOk("allowed_roles")),
-			ExternalOauthRsaPublicKey:                    d.Get("rsa_public_key").(string),
-			ExternalOauthRsaPublicKeyOk:                  isOk(d.GetOk("rsa_public_key")),
-			ExternalOauthRsaPublicKey2:                   d.Get("rsa_public_key_2").(string),
-			ExternalOauthRsaPublicKey2Ok:                 isOk(d.GetOk("rsa_public_key_2")),
-			ExternalOauthAudienceList:                    expandStringList(d.Get("audience_urls").(*schema.Set).List()),
-			ExternalOauthAudienceListOk:                  isOk(d.GetOk("audience_urls")),
-			ExternalOauthAnyRoleMode:                     snowflake.AnyRoleMode(d.Get("any_role_mode").(string)),
-			ExternalOauthAnyRoleModeOk:                   isOk(d.GetOk("any_role_mode")),
-			ExternalOauthScopeDelimiter:                  d.Get("scope_delimiter").(string),
-			ExternalOauthScopeDelimiterOk:                isOk(d.GetOk("scope_delimiter")),
-			ExternalOauthScopeMappingAttribute:           d.Get("scope_mapping_attribute").(string),
-			ExternalOauthScopeMappingAttributeOk:         isOk(d.GetOk("scope_mapping_attribute")),
-
-			Comment:   sql.NullString{String: d.Get("comment").(string)},
-			CommentOk: isOk(d.GetOk("comment")),
 		},
 	}
+}
 
-	stmt, err := manager.Create(input)
-	if err != nil {
-		return fmt.Errorf("couldn't generate create statement: %w", err)
-	}
-
+func ImportExternalOauthIntegration(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	logging.DebugLogger.Printf("[DEBUG] Starting external oauth integration import")
 	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
-	err = snowflake.Exec(db, stmt)
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+
+	integration, err := client.SecurityIntegrations.ShowByID(ctx, id)
 	if err != nil {
-		return fmt.Errorf("error executing create statement: %w", err)
+		return nil, err
 	}
 
-	d.SetId(ExternalOauthIntegrationID(&input.ExternalOauthIntegration3))
-
-	return ReadExternalOauthIntegration(d, meta)
-}
-
-// ReadExternalOauthIntegration implements schema.ReadFunc.
-func ReadExternalOauthIntegration(d *schema.ResourceData, meta interface{}) error {
-	manager, err := snowflake.NewExternalOauthIntegration3Manager()
+	integrationProperties, err := client.SecurityIntegrations.Describe(ctx, id)
 	if err != nil {
-		return fmt.Errorf("couldn't create external oauth integration builder: %w", err)
+		return nil, err
 	}
 
-	input := ExternalOauthIntegrationIdentifier(d.Id())
-
-	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
+	if err = d.Set("name", integration.Name); err != nil {
+		return nil, err
+	}
+	if err = d.Set("enabled", integration.Enabled); err != nil {
+		return nil, err
+	}
+	if oauthType, err := integration.SubType(); err == nil {
```

Comment:
Should we throw or continue on FindOne and SubType errors? 🤔

Reply:
Good question, but I did it like in other resources, so let's leave it for now.